### PR TITLE
Prepare flush etcd service

### PR DESCRIPTION
**Pull request by skovbakke**
Refactored parts of the boot script making it easier to add functionality. Then added functionality needed for the flush-etcd-service. 

* NODE_TYPE is no longer a binary variable. It takes three values: master-init, master-join, and worker.
* I fixed the indentation problems in `prepare_k8s_configs` and simplified it a bit. Note that there is an unused variable being set.
* The boot script is now a lot easier to work with.

### CODE REVIEW

**Comment by skovbakke** [comment]
Did some additional bugfixing. Everything should be working again.

**Review by jacobcrawford — CHANGES_REQUESTED** [review]
Looks good, but the readme and docs should be updated to reflect the changes.

File: scripts/prepare_k8s_configs.sh [hunk 1]
```diff
@@ -8,50 +8,36 @@ source $CONF
...
+			$(<$TEMPLATES_DIR/InitConfiguration.yaml )
+			---
+			$(<$TEMPLATES_DIR/ClusterConfiguration.yaml )
+		EOF
+		" > $OUTPUT_DIR/InitConfiguration.yaml
+	;;
+	master-join) 
+		export CONTROL_PLANE_REGISTRATION=$'controlPlane:\n    certificateKey: '${MASTER_CERTIFICATE_KEY}$'\n    localAPIEndpoint:\n      advertiseAddress: '${NODE_HOST_IP}$'\n      bindPort: 6443'
+	;;&
+	worker)
```
Comment (jacobcrawford) [inline scripts/prepare_k8s_configs.sh:31]: Remove unused variable.

**Review by jacobcrawford — CHANGES_REQUESTED** [review]

**Review by jacobcrawford — APPROVED** [review]
looks good.

**Pull request merged**